modules/Bio/EnsEMBL/Compara/Utils/JIRA.pm

```diff
@@ -22,235 +22,467 @@ package Bio::EnsEMBL::Compara::Utils::JIRA;
+}
 
-    my $ticket_summary = $ticket->{fields}->{summary};
-    $logger->info( 'Creating' . ' "' . $ticket_summary . '" ... ' );
+=head2 _request_password
 
-    # First check if the ticket already exists
-    if (my $existing_ticket_key = $parameters->{existing_tickets}->{ ($parameters->{division} // '') . '--' . $ticket_summary }) {
-        $logger->info(
-            'Skipped: This seems to be a duplicate of https://www.ebi.ac.uk/panda/jira/browse/'
-                . $existing_ticket_key
-                . "\n" );
-        return $existing_ticket_key;
-    }
+  Example     : my $password = $jira_adaptor->_request_password();
+  Description : Asks for the password of the given JIRA user. WARNING: the
+                password is returned as plain text, i.e. without encryption.
+  Return type : string
+  Exceptions  : none
 
-    my $endpoint = 'rest/api/latest/issue';
-    print "post_request($endpoint, $ticket, ....)\n";
-    # print Dumper $ticket;
-    my $response = $self->post_request( $endpoint, $ticket, $parameters, $logger );
-    
-    my $ticket_key = decode_json( $response->content() )->{'key'};
-    $logger->info( "Done\t" . $ticket_key . "\n\n" );
-    return $ticket_key;
+=cut
+
+sub _request_password {
+    my $self = shift;
+    my $user = $self->{_user};
+    print "\nPlease, type the JIRA password for user '$user':";
+    # Make password invisible on terminal
+    ReadMode('noecho');
+    my $password = ReadLine(0);
+    chomp $password;
+    # Restore typing visibility on terminal
+    ReadMode(0);
+    print "\n\n";
+    return $password;
 }
 
-=head2 post_request
+=head2 _create_new_ticket
 
-  Arg[1]      : String $endpoint - the request's endpoint
-  Arg[2]      : Hashref $content - the request's content
-  Arg[3]      : Hashref $parameters - parameters used for authorization
-  Arg[4]      : Bio::EnsEMBL::Utils::Logger $logger - object used for logging
-  Example     : my $response = post_request( $endpoint, $content, $parameters, $logger )
-  Description : Sends a POST request to the JIRA server
-  Return type : HTTP::Response object
+  Arg[1]      : hashref $ticket - a JIRA ticket to be created
+  Arg[2]      : hashref $existing_tickets - a hash of JIRA tickets with their
+                summary as key and their JIRA key as value
+  Example     : my $ticket_key = $jira_adaptor->_create_new_ticket( );
+  Description : Creates the JIRA ticket unless its summary is in
+                $existing_tickets, and returns its JIRA key
+  Return type : string
   Exceptions  : none
 
 =cut
 
-sub post_request {
-    my ( $self, $endpoint, $content, $parameters, $logger ) = @_;
+sub _create_new_ticket {
+    my ( $self, $ticket, $existing_tickets ) = @_;
+    my $issue_type = lc $ticket->{fields}->{issuetype}->{name};
+    my $summary = $ticket->{fields}->{summary};
+    $self->{_logger}->info(sprintf('Creating %s "%s" ... ', $issue_type, $summary));
+    my $ticket_key;
+    if (exists $existing_tickets->{$summary}) {
```

Comment:
I'm not sure the $existing_tickets check really belongs here. I think _create_new_ticket could be unaware of this - it should really _just_ create the ticket, IMO. Moving it would also help with my previous comment (to consider existing tickets even in dry_mode)